### PR TITLE
Attach AuthorizedUser records to existing Student/Professor records

### PR DESCRIPTION
**Pull request by justinkrol**
This is a follow up to #47, which added `AuthenticatedUser`.

This PR adds the functionality to link `AuthenticatedUser` records with `Student` or `Professor` records (which should be seeded as of now, maybe created by the Project Coordinator in the future).

## Create account
image

## Log in
image

## Welcome page

Yes, will need to make this look nicer.

### CODE REVIEW

**Comment by DerekStride** [comment]
I can't get registration or logging in to work, also there's no error messages.

**Comment by DerekStride** [comment]
The base of this PR is `wip-authentication` should it be `master`?

**Comment by DerekStride** [comment]
Also, I ran some 🎩 and registration wasn't working for me

**Comment by DerekStride** [comment]
Hmm, I set up some debuggers, and it was because I wasn't entering a valid student number. That should show up as an error on the page.

**Comment by justinkrol** [comment]
^ Yep working on it already, will have an update today

**Comment by justinkrol** [comment]
Updated

**Comment by JFleming4** [comment]
If i try to register with a student number i get this error
screen shot 2018-04-01 at 11 36 47 am
screen shot 2018-04-01 at 11 36 28 am

I logged in as the seeded Derek with password derek

**Comment by justinkrol** [comment]
You'll have to drop and recreate the schema.
The error in question here is because your db is already created, and the `student_number` column was previously a `double` (when you created it), but is changed in this PR to be a `String`.

```
# drop schema public cascade;
# create schema public;
```

File: src/main/java/app/controllers/AuthenticatedUserController.java [hunk 1]
```diff
@@ -25,7 +28,7 @@
     private AuthenticatedUserValidator authenticatedUserValidator;
 
     @Autowired
-    private UserDetailsService userDetailsService;
+    private StudentRepository studentRepository;
```
Comment (JFleming4) [inline src/main/java/app/controllers/AuthenticatedUserController.java:31]: You have the student repository but it isn't used

**Review by JFleming4 — APPROVED** [review]
Looks good

**Pull request merged**